### PR TITLE
weechat-matrix: substitute matrix_upload

### LINKED ISSUE #306551: `weechatScripts.weechat-matrix`: `/upload <file>` doesn't work
### Describe the bug
With the `weechat-matrix` script, `/upload <file>` errors with:
`Error with command 'Error with command 'matrix_upload'`
I'm not sure, but it could be because `weechat-matrix` [assumes it's installed under `PATH`.](https://github.com/poljar/weechat-matrix?tab=readme-ov-file#uploading-files)

### Steps To Reproduce
1. Install [weechat](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=weechat)
2. Install [weechatScripts.weechat-matrix](https://search.nixos.org/packages?channel=unstable&from=0&size=50&buckets=%7B%22package_attr_set%22%3A%5B%22weechatScripts%22%5D%2C%22package_license_set%22%3A%5B%5D%2C%22package_maintainers_set%22%3A%5B%5D%2C%22package_platforms%22%3A%5B%5D%7D&sort=relevance&type=packages&query=weechat)
3. Connect to a channel, and try `/upload /path/to/file` 

### Expected behavior
I expected it to properly upload the file.

### Screenshots
N/A

### Additional context
This is how I setup `weechat` in my dots.
```nix
    (weechat.override {
      configure = { ... }: {
        scripts = with pkgs.weechatScripts; [
          # ..
          weechat-matrix
          # ...
        ];
      };
    })
```

### Notify maintainers

@tilpner @emilazy

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.63, NixOS, 24.05 (Uakari), 24.05.20231127.8cfef69`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos, unstable"`
 - channels(bwkam): `"home-manager, nixpkgs"`
 - nixpkgs: `/home/bwkam/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by bwkam**
## Description of changes

Fixes #306551 

substituted the matrix_upload script with the right nix store path, so it actually works.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by getchoo — CHANGES_REQUESTED** [review]
lgtm! since we're reformatting the file though, could you use https://github.com/NixOS/nixfmt ?

**Review by eclairevoyant — APPROVED** [review]

File: pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix [hunk 1]
```diff
@@ -91,6 +91,7 @@ in buildPythonPackage {
     addToSearchPath program_PYTHONPATH $out/${python.sitePackages}
     patchPythonScript $out/share/matrix.py
     substituteInPlace $out/${python.sitePackages}/matrix/server.py --replace \"matrix_sso_helper\" \"$out/bin/matrix_sso_helper\"
+    substituteInPlace $out/${python.sitePackages}/matrix/uploads.py --replace \"matrix_upload\" \"$out/bin/matrix_upload\"
```
Comment (eclairevoyant) [inline pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix:94]:
```suggestion
    substituteInPlace $out/${python.sitePackages}/matrix/uploads.py --replace-fail \"matrix_upload\" \"$out/bin/matrix_upload\"
```

just noticed the deprecated --replace flag

Comment (bwkam) [reply pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix:94]: do I make that change across the whole file, or just this line?

Comment (eclairevoyant) [reply pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix:94]: whole file in a separate commit would be good.

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 308652` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>weechatScripts.weechat-matrix</li>
  </ul>
</details>

**Review by getchoo — APPROVED** [review]

**Review by eclairevoyant — APPROVED** [review]

**Review by itslychee — APPROVED** [review]

**Pull request merged**